internal/windevice/devicequery.go

```diff
@@ -22,8 +29,82 @@ const (
 	_DEVPROP_TYPE_STRING_LIST uint32 = (_DEVPROP_TYPE_STRING | _DEVPROP_TYPEMOD_LIST)
 
 	_DEVPKEY_LOCATIONPATHS_GUID = "a45c254e-df1c-4efd-8020-67d146a850e0"
+
+	_IOCTL_SCSI_GET_ADDRESS          = 0x41018
+	_IOCTL_STORAGE_GET_DEVICE_NUMBER = 0x2d1080
+)
+
+var (
+	// class GUID for devices with interface type Disk
+	// 53f56307-b6bf-11d0-94f2-00a0c91efb8b
+	devClassDiskGUID = guid.GUID{
+		Data1: 0x53f56307,
+		Data2: 0xb6bf,
+		Data3: 0x11d0,
+		Data4: [8]byte{0x94, 0xf2, 0x00, 0xa0, 0xc9, 0x1e, 0xfb, 0x8b},
+	}
 )
 
+// SCSI_ADDRESS structure used with IOCTL_SCSI_GET_ADDRESS.
+// defined here: https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/ntddscsi/ns-ntddscsi-_scsi_address
+type SCSIAddress struct {
+	Length     uint32
+	PortNumber uint8
+	PathId     uint8
+	TargetId   uint8
+	Lun        uint8
+}
+
+// STORAGE_DEVICE_NUMBER structure used with IOCTL_STORAGE_GET_DEVICE_NUMBER
+// https://learn.microsoft.com/en-us/windows/win32/api/winioctl/ns-winioctl-storage_device_number
+type StorageDeviceNumber struct {
+	DeviceType      uint32
+	DeviceNumber    uint32
+	PartitionNumber uint32
+}
+
+// getScsiAddress retrieves the SCSI address from a given disk handle
+func getScsiAddress(ctx context.Context, handle windows.Handle) (*SCSIAddress, error) {
```

Comment:
Since we aren't using ctx should it be removed?

Reply:
I deliberately added context in these methods because in future we may want to check against the deadline on this context and bail out if the syscall is taking longer.